### PR TITLE
router analytics -- close http response body

### PR DESCRIPTION
**Pull request by soamvasani**
This change is [Reviewable](https://reviewable.io/reviews/fission/fission/1180)

### CODE REVIEW

File: router/analytics.go [hunk 1]
```diff
@@ -56,6 +57,12 @@ func (a *Analytics) run() {
 			continue
 		}
 
-		_, _ = http.Post(a.url, "application/json", bytes.NewReader(msgbytes))
+		resp, err := http.Post(a.url, "application/json", bytes.NewReader(msgbytes))
+		if err != nil && resp != nil && resp.Body != nil {
+			// read body to end and close it; otherwise we
+			// leak some resources in the http transport
+			_, _ = ioutil.ReadAll(resp.Body)
```
Comment (life1347) [inline router/analytics.go]: actually, Close() drains data automatically. So maybe we can remove ReadAll here?

Comment (soamvasani) [reply router/analytics.go]: Didn't know that, thanks.  When is doEarlyClose set/unset?  (also, ioutil.Discard seems like a better way to write this)

Comment (life1347) [reply router/analytics.go]: It happens at the server side for connection reuse. https://github.com/golang/go/blob/2fd97ee25e8a623e460aff26b4c422dec9532654/src/net/http/server.go#L1007-L1009

File: router/analytics.go [hunk 2]
```diff
@@ -56,6 +56,10 @@ func (a *Analytics) run() {
 			continue
 		}
 
-		_, _ = http.Post(a.url, "application/json", bytes.NewReader(msgbytes))
+		resp, err := http.Post(a.url, "application/json", bytes.NewReader(msgbytes))
+		if err == nil && resp != nil && resp.Body != nil {
```
Comment (vishal-biyani) [inline router/analytics.go]: Nitpick, from Go source (https://golang.org/src/net/http/client.go) `When err is nil, resp always contains a non-nil resp.Body.` - so checking `resp.Body` is not nil is redundant.

**Pull request merged**